From 6d3323281e389c6e00faa7d32e74959fc9343b69 Mon Sep 17 00:00:00 2001
From: Alex Hung <alex.hung@canonical.com>
Date: Mon, 1 Oct 2018 22:55:36 +0800
Subject: [PATCH] fwts-frontend-text: an example to run tests based on
 chassis type

This is based on FWTS 18.09.00.

This dummy example checks SMBIOS chassis type and run tests accordingly.

Signed-off-by: Alex Hung <alex.hung@canonical.com>
---
 live-image/fwts-frontend-text | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/live-image/fwts-frontend-text b/live-image/fwts-frontend-text
index 3cded0ea..f794ccf1 100755
--- a/live-image/fwts-frontend-text
+++ b/live-image/fwts-frontend-text
@@ -260,6 +260,7 @@ if [ $? -ne 0 ]; then
 fi
 
 UNAME_PLATFORM=`(uname -m) 2>/dev/null` || UNAME_PLATFORM=unknown
+CHASSIS_TYPE=`(dmidecode -s chassis-type) 2>/dev/null` || CHASSIS_TYPE=unknown
 
 while true
 do
@@ -283,6 +284,7 @@ do
 			"ACPI" "ACPI Tests" off \
 			"UEFI" "UEFI Tests" off \
 			"SBBR" "ARM SBBR Tests" off \
+			"Chassis" "Chassis Tests" off \
 			"Recommended" "Recommended Tests" off \
 			"Recommended for IFV" "Recommended Tests for F/W Vendors" off \
 			"Selected" "Select Individual Tests" off \
@@ -320,6 +322,21 @@ do
 			do_test "--sbbr --skip-test=s3,s4" 'Server Base Boot Requirements (SBBR) Tests'
 			done_tests_finish
 			;;
+		'Chassis')
+			case "${CHASSIS_TYPE}" in
+				Desktop)
+					FWTS_OPTIONS="dmicheck fadt bgrt hest"
+					;;
+				Notebook)
+					FWTS_OPTIONS="acpi_ac acpi_battery acpi_lid"
+					;;
+				*)
+					FWTS_OPTIONS="--uefitests"
+					;;
+			esac
+			do_test "${FWTS_OPTIONS}" 'Running Chassis Tests'
+			done_tests_finish
+			;;
 		'Recommended')
 			case "${UNAME_PLATFORM}" in
 				ppc*)
-- 
2.17.1

